### PR TITLE
[gosrc2cpg] - Initial Lambda Handling

### PR DESCRIPTION
**Pull request by pandurangpatil**
1. Handling of basic lambda method (creation of Method and MethodRef node)
2. gitignore for dotnetastgen binaries

TODO:
1. Handling of Lambda TypeDecl along with struct type.
2. Adding more combinations of unit tests.
3. Handling of Closure use case.
4. Lambda invocation call node.

### CODE REVIEW

File: joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstForLambdaCreator.scala [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import io.shiftleft.codepropertygraph.generated.{ModifierTypes, NodeTypes}
+import ujson.Value
+
+trait AstForLambdaCreator(implicit withSchemaValidation: ValidationMode) { this: AstCreator =>
+
+  def astForFuncLiteral(funcLiteral: ParserNodeInfo): Seq[Ast] = {
+    val lambdaName = nextClosureName()
+    // if the top of the stack function is fake file level method node (which is checked with filename) then use the fully qualified package name as base fullname
+    val baseFullName = methodAstParentStack
+      .collectFirst({ case m: NewMethod if !m.fullName.endsWith(parserResult.filename) => m.fullName })
```
Comment (DavidBakerEffendi) [inline joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstForLambdaCreator.scala:18]: Maybe for future, but file-level method nodes should node start having the `MODULE` modifier, then you can check `.isModule` instead

**Review by DavidBakerEffendi — APPROVED** [review]

File: joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstForLambdaCreator.scala (hunk 1 above)
Comment (pandurangpatil) [reply joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstForLambdaCreator.scala:18]: noted

Comment (pandurangpatil) [reply joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstForLambdaCreator.scala:18]: @DavidBakerEffendi I added `MODULE` modifier for file-level method nodes. However, I cannot call `.isModule` on `NewMethod` object as it's a part of `AST` node.

Comment (pandurangpatil) [reply joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstForLambdaCreator.scala:18]: can you point me to any sample explaining what needs to be done. I will try to make the respective changes.

Comment (DavidBakerEffendi) [reply joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstForLambdaCreator.scala:18]: There are currently [modifier accessors here](https://github.com/joernio/joern/blob/master/semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/types/propertyaccessors/ModifierAccessors.scala#L44-L49), but it is a bit more complicated as I understand the complication of it being a `NewNode` now, so I don't think we can do anything here

**Pull request merged**